Formula/mactelnet.rb

```diff
@@ -0,0 +1,32 @@
+class Mactelnet < Formula
+  desc "MikroTik MAC-Telnet protocol, used in their RouterOS based products"
+  homepage "https://github.com/haakonnessjoen/MAC-Telnet"
+  url "https://github.com/haakonnessjoen/MAC-Telnet/archive/v0.4.4.tar.gz"
+  sha256 "2e6f041c0ff26597e6551cb564a0e41430a6ae183d31eb216493d862733f4c14"
+  license "GPL-2.0"
+
+  depends_on "gettext" => :build
+  depends_on "autoconf"
+  depends_on "automake"
+  depends_on "libtool"
+
+  def install
+    gettext = Formula["gettext"]
+    ENV.prepend_path "PATH", gettext.bin
+    ENV["LDFLAGS"] = "-L#{gettext.lib}"
+    ENV["CPPFLAGS"] = "-I#{gettext.include}"
+    system "./autogen.sh"
+    system "./configure",
+           "--disable-silent-rules",
+           "--disable-dependency-tracking",
+           "--prefix=#{prefix}",
+           "--with-libintl-prefix=#{gettext.include}"
+    system "make", "all", "install"
+  end
+
+  test do
+    system "#{bin}/macping", "-v"
+    system "#{bin}/mactelnet", "-v"
+    system "#{bin}/mactelnetd", "-v"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
```
septs@CBDesktop02 ~/Projects/MAC-Telnet (master) $ ./src/mactelnet --help
./src/mactelnet: illegal option -- -
./src/mactelnet: illegal option -- e
./src/mactelnet: option requires an argument -- p
Searching for MikroTik routers... Abort with CTRL+C.

IP              MAC-Address       Identity (platform version hardware) uptime
^C
septs@CBDesktop02 ~/Projects/MAC-Telnet (master) $ ./src/mactelnet -v
MAC-Telnet 0.4.4
septs@CBDesktop02 ~/Projects/MAC-Telnet (master) $ ./src/mactelnet
MAC-Telnet 0.4.4
Usage: ./src/mactelnet <MAC|identity> [-h] [-n] [-a <path>] [-A] [-t <timeout>] [-u <user>] [-p <password>] [-U <user>] | -l [-B] [-t <timeout>]
```

Sorry, it has no such option.

Reply:
Can't test, He works on the MAC layer

see https://en.wikipedia.org/wiki/OSI_model#Layer_2:_Data_Link_Layer